src/request-queue.ts

```diff
@@ -84,12 +84,14 @@ export class RequestQueue extends RequestQueueScaffold {
 
     unload(): void {
         clearTimeout(this._poller)
-        const requests = this._event_queue.length > 0 ? this.formatQueue() : {}
+        let requests = this._event_queue.length > 0 ? this.formatQueue() : {}
         this._event_queue.length = 0
-        for (const key in requests) {
-            const { url, data, options } = requests[key]
-            this.handlePollRequest(url, data, { ...options, transport: 'sendBeacon' })
-        }
+        Object.entries(requests)
+            // Always force events to be sent before recordings, as events are more important, and recordings are bigger and thus less likely to arrive
+            .sort((_, b) => b[1].url.indexOf('/e')) 
```

Comment:
This is too magical.

While it probably works (hard to say: no tests), the behaviour here is unspecified for browsers and hard to parse for a human. For example, some browser might choose to reverse all `/e` events, leading to mess.

I'd replace it with something more like:

```js
const requestValues = Object.values(requests)
const sortedRequests =  [
    ...requestValues.filter(r => r.url.indexOf('/e') === 0),
    ...requestValues.filter(r => r.url.indexOf('/e') !== 0),
]
```